### PR TITLE
14 - Design Top Nav Bar

### PR DESCRIPTION
**Pull request by lhuang408**
image

### CODE REVIEW

**Review by thedomcab — COMMENTED** [review]
we are using tailwind css. Coud you modify code to use it.

**Comment by lhuang408** [comment]
Needs work: Hamburger menu is not yet implemented in mobile

Desktop View
image

image

image


Mobile View (screens smaller than 900px)
image

File: frontend/src/components/header.js [hunk 1]
```diff
@@ -1,33 +1,122 @@
...
+  // }
+
+
+  return (
+    <header>
+      <div className={`${containerStyle}`}>
+        <div className="flex flex-row w-full justify-between [@media(min-width:900px)]:hidden relative">
+          <a href="/">
+            <img alt="hmcc logo" className={logoStyle} src={hmccLogo} />
+          </a>
```
Comment (thedomcab) [inline frontend/src/components/header.js:55]:
```suggestion
          <Link to="/">
            <img alt="hmcc logo" className={logoStyle} src={hmccLogo} />
          </Link>
```
Could you use the gatsby Link Component?  this is what been using for all our links.

File: frontend/src/components/header.js [hunk 2]
```diff
@@ -1,33 +1,122 @@
...
-    />
-  </header>
-);
+import hmccLogo from "../images/hmcc-ripple-white.svg";
+import mapPinLogo from "../images/icons/map-pin.svg";
+import dropDown from "../images/icons/dropdown.svg";
+import x from "../images/icons/X.svg";
```
Comment (thedomcab) [inline frontend/src/components/header.js:5]:
```suggestion
import x from "../images/icons/X.svg";
import { Link }  from gatsby;
```

File: frontend/src/components/header.js [hunk 3]
```diff
@@ -1,33 +1,175 @@
...
+                    key={`browseLink-${index}`}
+                    href={item.route}
+                    className={`${textStyle} ${
+                      item.route + "/" == path
+                        ? "font-extrabold py-2 px-4"
+                        : "font-medium py-2 px-4"
+                    } w-full text-left flex items-center pl-[56px] text-xl h-[75px] border-b-[0.5px] last:border-none`}
+                  >
+                    {item.title}
+                  </a>
```
Comment (thedomcab) [inline frontend/src/components/header.js:96]: Can you us Link Component?

File: frontend/src/components/header.js [hunk 4]
```diff
@@ -1,33 +1,175 @@
...
+                ))}
+                <div className={`${borderStyle} mt-6 w-[90vw] self-center`}>
+                  <a
+                    href="/new"
+                    className={`${textStyle} ${
+                      "/new/" == path ? "font-extrabold" : ""
+                    } font-bold hover:font-extrabold tracking-[0.96px]`}
+                  >
+                    I'm New
+                  </a>
```
Comment (thedomcab) [inline frontend/src/components/header.js:106]: Can you use Link Component?

File: frontend/src/components/header.js [hunk 5]
```diff
@@ -1,33 +1,175 @@
...
+                  {item.title}
+                </a>
+              ))}
+              <a
+                href="https://tangerang.hmcc.net/"
+                target="_blank"
+                className={`${textStyle} hover:bg-[#0C2966] py-2 px-4 tracking-[0.96px]`}
+              >
+                Tangerang
+              </a>
```
Comment (thedomcab) [inline frontend/src/components/header.js:139]: Can you use Link comment?

File: frontend/src/components/header.js [hunk 6]
```diff
@@ -1,33 +1,175 @@
...
+
+          <div className={borderStyle}>
+            <a
+              href="/new"
+              className={`${textStyle} ${
+                "/new/" == path ? "font-extrabold" : ""
+              } font-bold hover:font-extrabold tracking-[0.96px]`}
+            >
+              I'm New
+            </a>
```
Comment (thedomcab) [inline frontend/src/components/header.js:167]: Can you use Link component for anchor tag?

File: frontend/src/components/header.js [hunk 7]
```diff
@@ -1,33 +1,175 @@
...
+  const locationsList = [
+    { title: "Ann Arbor", route: "/" },
+    { title: "Austin", route: "https://austin.hmcc.net/" },
+    { title: "Detroit", route: "https://detroit.hmcc.net/" },
+    { title: "Hong Kong", route: "https://hongkong.hmcc.net/" },
+    { title: "Jakarta", route: "https://jakarta.hmcc.net/" },
+  ];
+
+  const [isOpen, setIsOpen] = React.useState(false);
+  const path = window.location.pathname;
```
Comment (thedomcab) [inline frontend/src/components/header.js:27]: here is only minore changes needes
```suggestion
  const path = typeof window !== "undefined" && window.location.pathname || "";
```

File: frontend/src/components/header.js [hunk 8]
```diff
@@ -1,33 +1,175 @@
...
+                  </Link>
+                ))}
+                <div className={`${borderStyle} mt-6 w-[90vw] self-center`}>
+                  <Link
+                    to="/new"
+                    className={`${textStyle} ${
+                      "/new/" == path ? "font-extrabold" : ""
+                    } font-bold hover:font-extrabold tracking-[0.96px]`}
+                  >
+                    I'm New
```
Comment (thedomcab) [inline frontend/src/components/header.js:105]:
```suggestion
                    I&apos;m New
```

File: frontend/src/components/header.js [hunk 9]
```diff
@@ -1,33 +1,175 @@
...
+          </div>
+
+          <div className={borderStyle}>
+            <Link
+              to="/new"
+              className={`${textStyle} ${
+                "/new/" == path ? "font-extrabold" : ""
+              } font-bold hover:font-extrabold tracking-[0.96px]`}
+            >
+              I'm New
```
Comment (thedomcab) [inline frontend/src/components/header.js:166]:
```suggestion
              I&apos;m New
```

**Pull request merged**